client.go

```diff
@@ -145,7 +145,12 @@ func (c *Client) request(ctx context.Context, method string, urlStr string, quer
 			return nil, fmt.Errorf("Retry request with 429 response failed after %d retries", failedAttempts)
 		}
 		// https://developers.notion.com/reference/request-limits#rate-limits
-		retryAfter := res.Header["Retry-After"][0]
+		retryAfterHeader := res.Header["Retry-After"]
+		if len(retryAfterHeader) == 0 {
+			return nil, fmt.Errorf("Retry-After header missing from Notion API response headers for 429 response")
```

Comment:
I think we should make a new error type for 429 and just return it here and [here](https://github.com/jomei/notionapi/pull/122/files#diff-4b667feae66c9d46b21b9ecc19e8958cf4472d162ce0a47ac3e8386af8bbd8cfR145)